### ISSUE TITLE
Alignment issue when moving form to a new line

### ISSUE THREAD

**Issue by knubie** [gpanders/nvim-parinfer#9]
Hi @gpanders , I've been using `nvim-parinfer` full time for the past few days now and I've encounter what appears to be a bug. I think the easiest way to describe it is to just show a short screen recording

https://user-images.githubusercontent.com/431251/153754948-8eb4921c-092b-4287-84b6-7608004f943c.mov

Here I'm moving the map form (`{...}`) to a new line using the `return` button in insert mode.

For reference here is the behavior on the parinfer demo page:

https://user-images.githubusercontent.com/431251/153755004-c4d6d1bf-fb65-49bc-93e4-2638d19da043.mov

I haven't tested this in `parinfer-lua` by itself so I can't say if it's a bug there or something specific to this plugin (or my set up).

I also noticed when hitting return multiple times, the form on the second line continues to outdent. Then when I hit `u` to undo, I get an error from the plugin:

https://user-images.githubusercontent.com/431251/153755173-765a32d4-7ece-4c18-9b7d-4ae44a3a9ee3.mov

**Comment by gpanders** [comment]
I've identified the cause of why this is happening: we track changes to the buffer in a callback and send those to `parinfer-lua`, which in turn uses those changes to determine how to indent subsequent lines. The issue is that when you add a new line we are receiving two separate "changes".

The first is the addition of the newline, which makes the following change:

```clojure
(
{:foo "bar"
  :bizz "baz"})
  
```

But of course if you have `'autoindent'` or `'smartindent'` or any other number of options set, Neovim will intelligently re-indent this line for you, causing a second "change"

```clojure
(
 {:foo "bar"
  :bizz "baz"})
```

When we send this change to `parinfer-lua`, it thinks that we have added a space to the beginning of line 2, so it indents the rest of the block with a space as well, resulting in

```clojure
(
 {:foo "bar"
   :bizz "baz"})
```

And each subsequent newline makes the problem worse.

I'm not sure what the fix is yet. I'm mostly writing up this description as notes for myself 😄

**Comment by knubie** [comment]
Ah ha, that makes perfect sense. Thanks for the detailed explanation.

When these kinds of double-change events occur, is it possible to send them to `parinfer-lua` sequentially so, for example, the first change happens, gets sent to `parinfer-lua`, then `parinfer-lua's` result gets fed back in when the next change is sent to `parinfer-lua`?

For example:

change `0` (starting state)
```clj
({:foo "bar"
  :bizz "baz"})
```

change `1` (after return in insert mode)
```clj
(
{:foo "bar"
  :bizz "baz"})
```

parinfer result `1-a`
```clj
(
{:foo "bar"
 :bizz "baz"})
```

change `2` (`'autoindent'`)
```clj
(
 {:foo "bar"
  :bizz "baz"})
```

Now at this point instead of sending change `1` and change `2` to `parinfer-lua` (which is what I presume to be happening), could we instead send result `1-a` and change `2` to `parinfer-lua`?

So in other words the changes stack sent to `parinfer-lua` would look like `[1-a, 2]` instead of `[1, 2]`.

**Comment by gpanders** [comment]
Ok it turns out the problem was actually how I was sending changes to parinfer. The offsets in the `on_bytes` handler were being interpreted incorrectly, so I was sending the wrong change objects which meant parinfer wasn't interpreting the changes properly. I fixed that in #10, which solves this issue in my testing. Can you give it a try?

**Comment by knubie** [comment]
Yep, works great now, thanks!

**Comment by gpanders** [comment]
Ok great. I’m going to daily drive those changes for a few days to make sure there aren’t any other bugs/regressions and then I’ll merge.
